README.md

```diff
@@ -286,14 +302,29 @@ When adding new tests:
 5. **Update VS Code config**: Add debug configurations for new test files
 ## Prometheus Monitoring
 
-This chart includes comprehensive Prometheus monitoring capabilities through the integrated [squid-exporter](https://github.com/boynux/squid-exporter) (upstream). The squid-exporter is built into the main Squid container, providing a single, consolidated image. The monitoring system provides detailed metrics about Squid's operational status, including:
+This chart includes comprehensive Prometheus monitoring capabilities through **two complementary exporters**:
+
+1. **Standard Squid Exporter** (Port 9301): Uses the [squid-exporter](https://github.com/boynux/squid-exporter) for general squid metrics - runs as separate container
+2. **Per-Site Metrics Exporter** (Port 9302): Custom exporter providing detailed per-site hit/miss ratios and performance metrics - **integrated into main squid container** using direct log piping
+
+### Container Architecture
+
+The deployment uses a **2-container pod architecture**:
+
+- **Main Container (`squid`)**: Runs squid proxy with integrated per-site exporter using direct piping: `squid | per-site-exporter`
+- **Sidecar Container (`squid-exporter`)**: Standard metrics exporter that connects to squid via HTTP API
+
+This consolidated approach eliminates the complexity of named pipes and volume mounts while maintaining clean separation of concerns.
```

Comment:
This is partially incorrect (it's really 3 containers) and doesn't provide any value. Please remove it.